src/Avalonia.Animation/Animatable.cs

```diff
@@ -66,19 +66,18 @@ public Transitions Transitions
         /// <param name="e">The event args.</param>
         protected override void OnPropertyChanged(AvaloniaPropertyChangedEventArgs e)
         {
-            if (e.Priority != BindingPriority.Animation && Transitions != null && _previousTransitions != null)
-            {
-                var match = Transitions.FirstOrDefault(x => x.Property == e.Property);
+            if (_transitions is null || _previousTransitions is null || e.Priority == BindingPriority.Animation) return;
 
-                if (match != null)
-                {
-                    if (_previousTransitions.TryGetValue(e.Property, out var dispose))
-                        dispose.Dispose();
+            var match = Transitions.FirstOrDefault(x => x.Property == e.Property);
```

Comment:
This line is responsible for like 90% `Func` allocations when launching ControlCatalog and it gets worse when you start clicking around. We should "UnLINQ" it.

![image](https://user-images.githubusercontent.com/2588062/64492203-f41a1c00-d271-11e9-8003-482339a9adfd.png)